src/analyses/osek.ml

```diff
@@ -1,1286 +0,0 @@
-                Hashtbl.add taskids (Cil.integer(intid)) (make_task objectname)
-              end
-            | x when Hashtbl.mem isrs (make_isr objectname) -> begin
-                if tracing then trace "osek" "Adding ID (%s) for isr %s\n" id objectname;
-                let intid = int_of_string id in
-                Hashtbl.add isrids (Cil.integer(intid)) (make_isr objectname)
-              end
-            | x when Hashtbl.mem resources objectname -> begin
-                if tracing then trace "osek" "Adding ID (%s) for resource %s\n" id objectname;
-                let intid = int_of_string id in
-                Hashtbl.add resourceids (Cil.integer(intid)) objectname
-              end
-            | x when Hashtbl.mem spinlocks objectname -> begin
-                if tracing then trace "osek" "Adding ID (%s) for spinlock %s\n" id objectname;
-                let intid = int_of_string id in
-                Hashtbl.add spinlockids (Cil.integer(intid)) objectname
-              end
-            | x when Hashtbl.mem events objectname -> begin
-                if tracing then trace "osek" "Adding ID (%s) for event %s\n" id objectname;
-                let intid = int_of_string id in
-                Hashtbl.add eventids (Cil.integer(intid)) objectname
-              end
-            | x when Hashtbl.mem alarms objectname -> begin
-                if tracing then trace "osek" "Adding ID (%s) for alarm %s\n" id objectname;
-                let intid = int_of_string id in
-                Hashtbl.add alarmids (Cil.integer(intid)) objectname
-              end
-            | _ -> if tracing then trace "osek" "No matching object found for %s\n" objectname
-          end;
-        end;
-        read_info ();
-      with
-      | End_of_file -> ()
-      | e -> raise e
-    in read_info ();
-    if tracing then trace "osek" "Done parsing IDs\n";
-    close_in input
-
-
-
-
-  module MyParam =
-  struct
-    module G = LockDomain.Priorities
-    let effect_fun ?write:(w=false) (ls: LockDomain.Lockset.t) =
-      let locks = LockDomain.Lockset.ReverseAddrSet.elements ls in
-      let prys = List.map names locks in
-      let staticprys = List.filter is_task_res prys in
-      let pry = resourceset_to_priority staticprys in
-      if pry = min_int then `Bot else `Lifted (Int64.of_int pry)
-    let check_fun = effect_fun
-  end
-
-  module M =
-  struct
-    include Mutex.MakeSpec (MyParam)
-
-    let extract_lock lock =
-      match lock with
-      | AddrOf (Var varinfo,NoOffset) -> LockDomain.Addr.from_var varinfo
-      | _ -> assert false
-
-    let special ctx lval f args =
-      (* simulate old mutex analysis special by emitting events directly, a la mutexEvents *)
-      begin match f.vname, args with
-        | ("GetResource" | "GetSpinlock"), [lock] ->
-          ctx.emit (Events.Lock (extract_lock lock, true))
-        | ("ReleaseResource" | "ReleaseSpinlock"), [lock] ->
-          ctx.emit (Events.Unlock (extract_lock lock))
```

Comment:
I guess these special function names can also be removed from `LibraryFunctions`.